app/lib/actions/katello/content_view/incremental_updates.rb

```diff
@@ -29,14 +35,38 @@ def plan(version_environments, composite_version_environments, content, dep_solv
             end
 
             if hosts.any? && !content[:errata_ids].blank?
-              errata = ::Katello::Erratum.with_identifiers(content[:errata_ids])
-              hosts = hosts.where(:id => ::Katello::Host::ContentFacet.with_applicable_errata(errata).pluck(:host_id))
               plan_action(::Actions::BulkAction, ::Actions::Katello::Host::Erratum::ApplicableErrataInstall, hosts, content[:errata_ids])
             end
             plan_self(:version_outputs => output_for_version_ids)
           end
         end
 
+        def find_applicable_hosts_and_smart_proxies(hosts, content)
+          smart_proxies = []
+          if hosts.any? && !content[:errata_ids].blank?
+            errata = ::Katello::Erratum.with_identifiers(content[:errata_ids])
+            hosts = hosts.where(:id => ::Katello::Host::ContentFacet.with_applicable_errata(errata).pluck(:host_id))
```

Comment:
Hey @pmoravec I'm trying to understand why you added a second lookup/filter for hosts when the action has already been told which hosts to be concerned with. It appears to me that the controller which invokes this action has already determined the applicable hosts here: https://github.com/Katello/katello/blob/a1c9280067607999cae43bab89b53ba870856b76/app/controllers/katello/api/v2/content_view_versions_controller.rb#L162-L176

In other words: I have a suspicion that the `.where` here does not actually do any extra filtering and would return the same set of hosts

Reply:
I see the code was written that way before, and it's probably just a safeguard

Reply:
Indeed, I just moved that filter from original line 33 here / where it is applicable in new code. But it really seems it is a redundant filter as you pointed out, `calculate_hosts_for_incremental` seems to be executed everytime before this.

I am leaving the filter as a safe-guard now as a conservative/minimalistic change - if you think it is really safe to remove it, I can include the removal in my PR.